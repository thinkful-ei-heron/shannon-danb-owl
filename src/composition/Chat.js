import React from 'react';
import ChatItem from './ChatItem'
import STORE from '../STORE'


function Chat() {
  const chatEvents = STORE.chatEvents
  const participants = STORE.participants
  return (
    <div className="chat-window">
			<div className="stream">
    {chatEvents.map((item) =>
        <ChatItem name={participants[item.id]} message={item.message} avatar='g' time={item.timestamp} type={item.type} />
    )}
      </div>
      <div className="input">
        <input type="text" />
      </div>
    </div>
  );
}

export default Chat;

