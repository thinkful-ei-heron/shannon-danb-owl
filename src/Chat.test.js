import React from 'react';
import ReactDOM from 'react-dom';
import Chat from './composition/Chat';
import renderer from 'react-test-renderer';


describe ('rendering Sidebar', () => {
  it ('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Chat />, div);
        ReactDOM.unmountComponentAtNode(div);
      
  });

  it ('renders correctly', () => {
      const tree = renderer
      .create(<Chat />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });
})

