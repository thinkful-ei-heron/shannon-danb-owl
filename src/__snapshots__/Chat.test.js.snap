// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering Sidebar renders correctly 1`] = `
<div
  className="chat-window"
>
  <div
    className="stream"
  >
    <div>
      <div>
        <img
          alt=""
          className="message-img"
        />
      </div>
      <div>
        <header
          className="message-header"
        >
          <h5>
            Axe Kubicka
          </h5>
          <p>
            4:48
          </p>
        </header>
        <div
          className="message-body"
        >
          Hello world
        </div>
      </div>
    </div>
    <div>
      Caty Flucker
       + 
      thumbs-up
    </div>
    <div>
      Frank Runciman
       + 
      thumbs-down
    </div>
    <div>
      Koren Templeton
       + 
      raise-hand
    </div>
    <div>
      Koren Templeton
       + 
      clap
    </div>
    <div>
      Koren Templeton
       + 
      join
    </div>
    <div>
      Ashla Attwool
       + 
      leave
    </div>
    <div>
      Axe Kubicka
       + 
      join-stage
    </div>
    <div>
      Axe Kubicka
       + 
      leave-stage
    </div>
  </div>
  <div
    className="input"
  >
    <input
      type="text"
    />
  </div>
</div>
`;
